R/begin.R

```diff
@@ -16,14 +17,31 @@
 begin = function(
   wd = NULL, 
   load = c( 'magrittr', 'dplyr' ),
+  keep = NULL,
   scipen = FALSE,
   verbose = TRUE,
   repos = 'http://cran.us.r-project.org'
 ){
   
   # clear workspace.
-  rm( list = ls( all.names = TRUE, envir = parent.frame() ), envir = parent.frame() )
-  if( verbose ) cat( 'Cleared variables from environment. \n' )
+  envobjs = ls( all.names = TRUE, envir = parent.frame() )
+  drop = setdiff( envobjs, keep )
+  # check if attempting to keep any objects not in environment
+  if( length( keep ) > 0 ){
+    missing = setdiff( keep, envobjs )
+    if( length( missing ) > 0 ){ 
+      stop( 
+        paste0( 'Attempting to keep objects not in environment: [', 
+        paste0( missing, collapse = ', ' ), ']. ' 
```

Comment:
ooh great error handling here.

Reply:
FWIW I would do this:
```
stop(glue('
    Attempted to keep objects not in the environment: [{cc(missing, sep = ", ")}]. Error easyr::begin E911.
')) 
```
Glue is nice to keep things simpler and it helps to add the function name for when RStudio doesn't give the file name.  I use the current time as the error number for ease.

Reply:
Ok. I was trying to rely on base functions only so I didn't add new dependencies to easyr. However, I'm betting easyr already relies on glue so I've made the change.

Reply:
Also, should a package rely on itself? Asking because cc is an easyr function.

Reply:
That's a good question, you are correct that it is a good practice to use base functions. Searching the other code, cc is used  lot without the :: required by CRAN for external function calls so I think it's fine to use it. That said, I think it'd be fine to use paste0 if you'd prefer.